diffusers/Using-IP-Adapter.ipynb

```diff
@@ -0,0 +1,304 @@
+{
```

Comment:
<p>Replace with:</p><p> </p>

<pre class="ql-syntax" spellcheck="false"># IP Adapter Negative Noise

Diffusers pipelines are fully integrated with IP-Adapter, which allows you to prompt the diffusion model with an image. However, it does not support negative image prompts (there is no&nbsp;<code>negative_ip_adapter_image</code>&nbsp;argument) the same way it supports negative text prompts. When you pass an&nbsp;<code>ip_adapter_image</code>,&nbsp;it will create a zero-filled tensor as a negative image. This script shows you how to create a negative noise from&nbsp;<code>ip_adapter_image</code>&nbsp;and use it to significantly improve the generation quality while preserving the composition of images.

[cubiq](https://github.com/cubiq)&nbsp;initially developed this feature in his&nbsp;[repository](https://github.com/cubiq/ComfyUI_IPAdapter_plus). The community script was contributed by&nbsp;[asomoza](https://github.com/Somoza). You can find more details about this experimentation&nbsp;[this discussion](https://github.com/huggingface/diffusers/discussions/7167).
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/notebooks/pull/525/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>